client/src/features/NetworkGraph/NetworkGraph.tsx

```diff
@@ -11,10 +11,11 @@ import type { Simulation } from "d3";
 import type { Link, Node, Group, HubLink } from "../../types/network-graph.types.ts";
 
 export const NetworkGraph = () => {
-    const { data } = useDummyData();
+    const data = useDummyData();
+
     const canvasRef = useRef<HTMLCanvasElement>(null);
 
-    const centers = (w: number, h: number) => ({
+    const nodeGroupCenters = (w: number, h: number) => ({
```

Comment:
When you remove the field `group` from the json, please be sure to rename these variables consistently or it will be very confusing.